spec/models/manageiq/providers/hawkular/middleware_manager/refresh_parser_spec.rb

```diff
@@ -57,4 +57,23 @@
       expect(parser.send(:parse_datasource, server, datasource, config)).to eq(parsed_datasource)
     end
   end
+
+  describe 'alternate_machine_id' do
+    it 'should transform machine ID to dmidecode BIOS UUID' do
+      # the /etc/machine-id is usually in downcase, and the dmidecode BIOS UUID is usually upcase
+      # the alternate_machine_id should *just* swap digits, it should not handle upcase/downcase.
+      # 33D1682F-BCA4-4B4C-B19E-CB47D344746C is a real BIOS UUID retrieved from a VM
+      # 33d1682f-bca4-4b4c-b19e-cb47d344746c is what other providers store in the DB
+      # 2f68d133a4bc4c4bb19ecb47d344746c is the machine ID for the BIOS UUID above
+      # at the Middleware Provider, we get the second version, while the first is usually used by other providers
+      machine_id = '2f68d133a4bc4c4bb19ecb47d344746c'
+      expected = '33d1682f-bca4-4b4c-b19e-cb47d344746c'
```

Comment:
if the code added adds support for more than just rhevm/rhel ([which it currently does](https://github.com/ManageIQ/manageiq/pull/9296#discussion-diff-67717632R65)), then the tests should cover it.
if the code that goes in will be only for rhev/rhel, then I agree - testing only that particular thing at the moment.